DESCRIPTION: This patch is for our case when we want to have a
static Nodes in the cluster, managed by vSphere cloud provider.
~~~
Index: pkg/cloudprovider/vsphere/instances.go
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/pkg/cloudprovider/vsphere/instances.go b/pkg/cloudprovider/vsphere/instances.go
--- a/pkg/cloudprovider/vsphere/instances.go	(revision 69cb68d0e885ee9573f68efddaf040caf3534552)
+++ b/pkg/cloudprovider/vsphere/instances.go	(revision 1fa841dbb4017f0ec82e2093072db211c73b7f41)
@@ -152,6 +152,10 @@
 func (i *instances) InstanceExistsByProviderID(ctx context.Context, providerID string) (bool, error) {
 	klog.V(4).Info("instances.InstanceExistsByProviderID() called with ", providerID)

+	if providerID == "static://" {
+		return true, nil
+	}
+
 	// Check if node has been discovered already
 	uid := GetUUIDFromProviderID(providerID)
 	if _, ok := i.nodeManager.nodeUUIDMap[uid]; ok {
